File: sail_server/router/file_storage.py
# -*- coding: utf-8 -*-
# @file file_storage.py
# @brief File Storage Router
# @author sailing-innocent
# @date 2026-03-14
# @version 1.0
# ---------------------------------

from litestar import Router
from sail_server.controller.file_storage import FileStorageController

router = Router(
    path="/file-storage",
    route_handlers=[FileStorageController],
)

File: site/src/lib/data/file_storage.ts
/**
 * @file file_storage.ts
 * @brief File Storage Data Types
 * @author sailing-innocent
 * @date 2026-03-14
 */

export interface FileUploadResponse {
  filename: string
  original_name: string
  size: number
  message: string
}

export interface FileInfo {
  filename: string
  original_name: string
  size: number
  created_at: string
  updated_at: string
}

export interface FileListResponse {
  files: FileInfo[]
  total: number
}

export interface FileDeleteResponse {
  filename: string
  message: string
}

export interface FileContentResponse {
  filename: string
  original_name: string
  content: string
  size: number
}
